src/internal/Subscription.ts

```diff
@@ -144,10 +144,28 @@ export class Subscription implements SubscriptionLike {
   }
 }
 
+// Even though Subscriptoin only conditionally implements `Symbol.dispose`
+// if it's available, we still need to declare it here so that TypeScript
+// knows that it exists on the prototype when it is available.
+export interface Subscription {
+  [Symbol.dispose](): void;
+}
+
+if (typeof Symbol.dispose === 'symbol') {
+  Subscription.prototype[Symbol.dispose] = Subscription.prototype.unsubscribe;
+}
+
 function execFinalizer(finalizer: Unsubscribable | (() => void)) {
   if (isFunction(finalizer)) {
     finalizer();
   } else {
     finalizer.unsubscribe();
   }
 }
+
+// Ensure that `Symbol.dispose` is defined in TypeScript
+declare global {
```

Comment:
Same as above. Also I suspect this will be removed in soon-released versions of TypeScript

Reply:
If I'm not wrong, it will be added in 5.2 (next release), as they are also adding support for `using`. Merged a few weeks ago https://github.com/microsoft/TypeScript/pull/54505

Source code of disposable interfaces: https://github.com/microsoft/TypeScript/blob/main/src/lib/esnext.disposable.d.ts#L7